lib/src/mock_field_value_platform.dart

```diff
@@ -1,22 +1,31 @@
 import 'package:cloud_firestore_platform_interface/cloud_firestore_platform_interface.dart';
+import 'package:fake_cloud_firestore/src/fake_server_time_provider.dart';
 import 'package:fake_cloud_firestore/src/util.dart';
 import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 
 abstract class FakeFieldValue {
   const FakeFieldValue();
 
   static const delete = FieldValueDelete();
-  static const serverTimestamp = FieldValueServerTimestamp();
+  static final serverTimestamp = FieldValueServerTimestamp();
 
   void updateDocument(Map<String, dynamic> document, String key);
 }
 
 class FieldValueServerTimestamp extends FakeFieldValue {
-  const FieldValueServerTimestamp();
+  FakeServerTimeProvider? _fakeServerTimeProvider;
+
+  Timestamp get now => _fakeServerTimeProvider?.now ?? Timestamp.now();
+
+  void setTimeProvider(FakeServerTimeProvider? fakeServerTimeProvider) {
```

Comment:
How about using a getter setter?

```dart
set clock(Clock clock) {
  this._clock = clock;
}
```

Reply:
let me think, I was thinking about how the `FieldValueServerTimestamp` can get the clock from FakeFirebaseFirestore